spec/controllers/articles_controller_spec.rb

```diff
@@ -54,87 +58,132 @@
   end
 
   describe "#create" do
-    let(:valid_params) {
-      { title: "Article Title", body: "Article body" }
-    }
+    let(:valid_params) { FactoryBot.attributes_for(:article) }
 
-    it "should create an article when given correct params" do
-      expect {
-        post :create, params: { article: valid_params }
-      }.to change(Article, :count).by(1)
+    context "is valid" do
+      it "should create an article when given correct params" do
+        expect {
+          post :create, params: { article: valid_params }
+        }.to change(Article, :count).by(1)
+      end
 
-      assert_redirected_to articles_url
+      it "should redirect to the articles page" do
+        post :create, params: { article: valid_params }
+        assert_redirected_to articles_url
+      end
     end
 
-    it "should not article when given incorrect params" do
-      expect {
-        post :create, params: { article: { title: "article title ", body: nil } }
-      }.to_not change(Article, :count)
+    context "is invalid" do
+      it "should not create an article when given incorrect params" do
+        expect {
+          post :create, params: { article: { title: valid_params[:title], body: nil } }
+        }.to_not change(Article, :count)
+      end
 
-      expect(response).to render_template(:new)
+      it "should render the new template" do
+        post :create, params: { article: { title: valid_params[:title], body: nil } }
+        expect(response).to render_template(:new)
+      end
     end
   end
 
   describe "#edit" do
-    before do
-      @article = Article.create(title: "Article Title", body: "Article Body")
-    end
-
-    it "should respond with 200 for an existing article" do
-      get :edit, params: { id: @article.id }
-      aggregate_failures do
-        expect(response).to be_successful
-        expect(response).to have_http_status "200"
+    let(:article) { FactoryBot.create(:article) }
+
+    context "is valid" do
+      it "should respond with 200 for an existing article" do
+        get :edit, params: { id: article.id }
+        aggregate_failures do
+          expect(response).to be_successful
+          expect(response).to have_http_status "200"
+        end
       end
-    end
 
-    it "should respond with 404 for a nonexistent article" do
-      get :edit, params: { id: 100 }
-      expect(response).to_not be_successful
-      expect(response).to have_http_status "404"
+      it "should render the edit template" do
+        get :edit, params: { id: article.id }
+        expect(response).to render_template(:edit)
+      end
     end
 
-    it "should render the show template" do
-      get :edit, params: { id: @article.id }
-      expect(response).to render_template(:edit)
+    context "is invalid" do
+      it "should respond with 404 for a nonexistent article" do
+        get :edit, params: { id: 100 }
+        aggregate_failures do
```

Comment:
This is a really good tool to help speed up tests, but I can make it a little more difficult to pinpoint where exactly something is failing. If you don't need performance right now I would go with having clarity